osf/migrations/0135_migrate_registrations_to_osf_registries_provider.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-07-18 13:51
+from __future__ import unicode_literals
+import logging
+from itertools import chain
+
+from django.db import migrations
+from django.db.models import Q
+from django.core.paginator import Paginator
+from django_bulk_update.helper import bulk_update
+
+logger = logging.getLogger(__file__)
+
+def migrate_existing_registrations_into_osf_registries(state, schema):
+    AbstractProvider = state.get_model('osf', 'abstractprovider')
+    AbstractNode = state.get_model('osf', 'abstractnode')
+    DraftRegistration = state.get_model('osf', 'draftregistration')
+
+    try:
+        osf_registries = AbstractProvider.objects.get(_id='osf', type='osf.registrationprovider')
+    except AbstractProvider.DoesNotExist:
+        # Allow test / local dev DBs to pass
+        logger.warn('Unable to find OSF Registries provider - assuming test environment.')
+        pass
+    else:
+        public_registrations_list = AbstractNode.objects.filter(
```

Comment:
Lastly, because the `provider` is set in `register_node` (upon registration creation), all existing registrations should have it set. Viewability permissions (embargoed, withdrawn, etc.) will still be enforced in the API and at Elastic ingress.

Therefore, the meat of this migration can be replaced with two lines:
``` python
DraftRegistration.objects.update(provider_id=osf_registries.id)
AbstractNode.objects.filter(type='osf.registration').update(provider_id=osf_registries.id)
# The latter could also be done with Registration.objects.update, but this is less efficient because
# TypeModels forces pk evaluation and then does `WHERE id IN <list of ~40k ids>`, whereas
# Filtering by type translates to `WHERE type = 'osf.registries'`. Both are indexed, but type skips
# the primary key evaluation step.
```